src/uwtools/utils/tasks.py

```diff
@@ -70,4 +70,7 @@ def symlink(target: Path, linkname: Path):
     yield asset(linkname, linkname.exists)
     yield existing(target)
     linkname.parent.mkdir(parents=True, exist_ok=True)
-    os.symlink(src=target, dst=linkname)
+    os.symlink(
+        src=target if target.is_absolute() else target.relative_to(linkname.parent, walk_up=True),
+        dst=linkname,
+    )
```

Comment:
This update supports target files with relative paths. For example (and this is taken from the dynamic docs for the `file link` tool/action), given `src/` directory
```
src
├── bar
└── foo
```
and `config.yaml`
```
foo: src/foo
subdir/bar: src/bar
```
command `uw file link --target-dir dst --config-file config.yaml` would previously have given us
```
dst
├── foo -> src/foo
└── subdir
    └── bar -> src/bar
```
which is wrong. With the change, we get
```
dst
├── foo -> ../src/foo
└── subdir
    └── bar -> ../../src/bar
```
which is correct.

Reply:
Why not use the `pathlib` methods for both of these? Is there something I'm missing about using `target.relative_to(linkname.parent)`?

Reply:
The `pathlib` solution doesn't work, unfortunately. From [the docs](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.relative_to) and in re: the function signature
```
PurePath.relative_to(other, walk_up=False)
```
> When `walk_up` is false (the default), the `path` must start with other.
But `walk_up` was added in 3.12, and the `False` behavior was the previous behavior, so
```
>>> Path("/foo/bar/baz").relative_to("/foo")
PosixPath('bar/baz')
```
is ok but
```
>>> Path("/foo/bar/baz").relative_to("/foo/qux")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.10/pathlib.py", line 818, in relative_to
    raise ValueError("{!r} is not in the subpath of {!r}"
ValueError: '/foo/bar/baz' is not in the subpath of '/foo/qux' OR one path is relative and the other is absolute.
```
is not. Compare to
```
>>> os.path.relpath("/foo/qux", "/foo/bar/baz")
'../../qux'
```
The release notes for Python 3.12 say
> Add `walk_up` optional parameter to [`pathlib.PurePath.relative_to()`](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.relative_to) to allow the insertion of .. entries in the result; this behavior is more consistent with [`os.path.relpath()`](https://docs.python.org/3/library/os.path.html#os.path.relpath).

Reply:
So, we can switch to the `pathlib` solution with `walk_up` (which does work, I tested it) when we only support Python 3.12+, but that'll be years from now.